sycl/unittests/Extensions/CommandGraph.cpp

```diff
@@ -624,8 +623,11 @@ TEST_F(CommandGraphTest, SubGraph) {
   ASSERT_EQ(sycl::detail::getSyclObjImpl(MainGraph)->MRoots.size(), 1lu);
   ASSERT_EQ(sycl::detail::getSyclObjImpl(Node1MainGraph)->MSuccessors.size(),
             1lu);
-  ASSERT_EQ(sycl::detail::getSyclObjImpl(Node1MainGraph)->MSuccessors.front(),
-            sycl::detail::getSyclObjImpl(Node1Graph));
+  // Subgraph nodes are duplicated when inserted to parent graph.
+  // we thus check the node type only.
+  ASSERT_TRUE(
+      *(sycl::detail::getSyclObjImpl(Node1MainGraph)->MSuccessors.front()) ==
+      *(sycl::detail::getSyclObjImpl(Node1Graph)));
```

Comment:
This comment is a bit confusing, since doesn't this operator not only check the type but kernel names in this case also?